### PR TITLE
Fix #8670 - Example IDF File not valid per IDD: UnitaryHybridAC_DedicatedOutsideAir.idf

### LINKED ISSUE #8670: Example IDF File not valid per IDD: UnitaryHybridAC_DedicatedOutsideAir.idf
Issue overview
--------------

### UnitaryHybridAC_DedicatedOutsideAir.idf

In `ZoneHVAC:HybridUnitaryHVAC` (the only one), Mode 10 has only 24 values entered instead of 25, it's missing the `Supply Air Temperature Lookup Table Name`

```diff
    ,                        !- Mode 10 Name
-   ,                        !- Mode 10 Supply Air Temperature Lookup Table Name
    ,                        !- Mode 10 Supply Air Humidity Ratio Lookup Table Name
    ,                        !- Mode 10 System Electric Power Lookup Table Name
```

Note: I tracked it down to the origin of the file, it's always been the case (https://raw.githubusercontent.com/NREL/EnergyPlus/387d18b5017dfe39a1be52138a040589b81ec215/datasets/UnitaryHybridUnitTest_MuntersEPX5000.idf)

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version): all
 - Version of EnergyPlus (if using an intermediate build, include SHA): develop, 1221990e789655f7812ceee0f76fd1d54f0d11a6
 - https://github.com/NREL/OpenStudio/issues/4268

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

 - Fix #8670 - Example IDF File not valid per IDD: UnitaryHybridAC_DedicatedOutsideAir.idf

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: testfiles/UnitaryHybridAC_DedicatedOutsideAir.idf [hunk 1]
```diff
@@ -857,6 +857,7 @@
     0.1,                     !- Mode 9 Minimum Supply Air Mass Flow Rate Ratio
     1,                       !- Mode 9 Maximum Supply Air Mass Flow Rate Ratio
     ,                        !- Mode 10 Name
+    ,                        !- Mode 10 Supply Air Temperature Lookup Table Name
```
Comment (mjwitte) [inline testfiles/UnitaryHybridAC_DedicatedOutsideAir.idf]: @jmarrec Energy+.idd is also missing this field.
```
  A110, \field Mode 10 Name
        \type alpha
        \retaincase
        \note Enter a name for Mode 10.
        \type alpha
        \retaincase
        \note Enter the name of the Supply Air Temperature Lookup Table for Mode 10.
        \note Units for lookup table values should be in C.
        \note If this field is blank, Mode 10 will not be considered for any time step that requires ventilation, heating, cooling, humidification, or dehumidification.
  A111, \field Mode 10 Supply Air Humidity Ratio Lookup Table Name
        \type object-list
        \object-list MultivariateFunctions
        \note Enter the name of the Supply Air Humidity Ratio Lookup Table for Mode 10.
        \note Units for lookup table values should be in kgWater/kgDryAir.
        \note If this field is blank, Mode 10 will not be considered for any time step that requires ventilation, heating, cooling, humidification, or dehumidification.
  A112, \field Mode 10 System Electric Power Lookup Table Name
```

Comment (jmarrec) [reply testfiles/UnitaryHybridAC_DedicatedOutsideAir.idf]: That would explain it... I'll add it

Comment (jmarrec) [reply testfiles/UnitaryHybridAC_DedicatedOutsideAir.idf]: Thanks for the catch @mjwitte 

Added IDD field in https://github.com/NREL/EnergyPlus/pull/8671/commits/670e1962dba7cc919ddaba609a4f31d6443d6e50.

This object really would be better off being an actual Extensible object... It is 5000 lines long, 660 fields.

**Comment by Myoldmopar** [comment]
@mjwitte if you are satisfied with this, you don't need to wait on all of CI to finish.  I believe this is the last thing that can go in prior to the wrap up PRs.

**Comment by mjwitte** [comment]
@Myoldmopar I'm a little puzzled why there weren't any diffs from the first change to idf only. Maybe it doesn't use these higher modes?

**Comment by jmarrec** [comment]
> @Myoldmopar I'm a little puzzled why there weren't any diffs from the first change to idf only. Maybe it doesn't use these higher modes?

I think it's because the Mode 6 is plain empty. On https://github.com/NREL/EnergyPlus/blob/a22993331b509d0adac7f79776c1e2591235d706/src/EnergyPlus/HybridUnitaryAirConditioners.cc#L636

Numberofoperatingmodes = 6

**Comment by jmarrec** [comment]
Yup, https://github.com/NREL/EnergyPlus/blob/a22993331b509d0adac7f79776c1e2591235d706/src/EnergyPlus/HybridUnitaryAirConditioners.cc#L628-L634

So when it reaches 

```
    ,                        !- Mode 6 Name
```

lAlphaBlanks(i) is true, so break. The rest is never read

Really, None of the others are actually filled out. The IDF snippet should be truncated at the end of Mode 5. This looks like a case of IDF Editor filling the default fields out, because this object isn't extensible.

**Comment by jmarrec** [comment]
I cut the IDF object at end of Mode 5 in https://github.com/NREL/EnergyPlus/pull/8671/commits/24ad155b966859aa3764fa3fb80c2c46f6d14931 @mjwitte

**Review by mjwitte — APPROVED** [review]
@jmarrec I was just coming to the same conclusion that it's not using most of that object. It also doesn't even look at `A8, \field Method to Choose Controlled Inputs and Part Runtime Fraction`.
Tested the revised file locally. All good.

**Pull request merged**